### PR TITLE
Break db_stress_tool.cc to a list of source files

### PR DESCRIPTION
**Pull request by siying**
Summary:
db_stress_tool.cc now is a giant file. In order to main it easier to improve and maintain, break it down to multiple source files.
Most classes are turned into their own files. Separate .h and .cc files are created for gflag definiations. Another .h and .cc files are created for some common functions. Some test execution logic that is only loosely related to class StressTest is moved to db_stress_driver.h and db_stress_driver.cc. All the files are located under db_stress_tool/. The directory name is created as such because if we end it with either stress or test, .gitignore will ignore any file under it and makes it prone to issues in developements.

Test Plan:
Build under GCC7 with and without LITE on using GNU Make. Build with GCC 4.8. Build with cmake with -DWITH_TOOL=1

### CODE REVIEW

File: tools/stress_test_base.h [hunk 1]
```diff
@@ -0,0 +1,7 @@
+//  Copyright (c) 2011-present, Facebook, Inc.  All rights reserved.
```
Comment (riversand963) [inline tools/stress_test_base.h]: Is this file being used?

File: TARGETS [hunk 2]
```diff
@@ -402,9 +402,17 @@ cpp_library(
 cpp_library(
     name = "rocksdb_stress_lib",
     srcs = [
+        "db_stress_tool/batched_ops_stress.cc",
```
[the remark this answers is not in the document]
Comment (siying) [reply TARGETS]: Actually, the file is auto generated by buckifer script.

File: db_stress_tool/batched_ops_stress.cc [hunk 3]
```diff
@@ -0,0 +1,342 @@
...
+// found in the LICENSE file. See the AUTHORS file for names of contributors.
+
+#ifdef GFLAGS
+#include "db_stress_tool/db_stress_common.h"
+
+namespace rocksdb {
+class BatchedOpsStressTest : public StressTest {
+ public:
+  BatchedOpsStressTest() {}
+  virtual ~BatchedOpsStressTest() {}
```
[the remark this answers is not in the document]
Comment (siying) [reply db_stress_tool/batched_ops_stress.cc]: Sure. I'll do it as a follow-up.

File: tools/stress_test_base.h (hunk 1 above)
Comment (siying) [reply tools/stress_test_base.h]: Hmm. It's not. Let me remove it.

**Pull request closed without merging**